plugins/components/conversionlayer.go

```diff
@@ -397,39 +397,53 @@ func fillFlagMaps(c *Context, baseContext *cli.Context, originalFlags []Flag) er
 	// Loop over all plugin's known flags.
 	for _, flag := range originalFlags {
 		if stringFlag, ok := flag.(StringFlag); ok {
-			finalValue, err := getValueForStringFlag(stringFlag, baseContext.String(stringFlag.Name))
+			finalValue, skip, err := getValueForStringFlag(stringFlag, baseContext)
 			if err != nil {
 				return err
 			}
+			if skip {
+				continue
+			}
 			c.stringFlags[stringFlag.Name] = finalValue
-			continue
 		}
 
 		if boolFlag, ok := flag.(BoolFlag); ok {
-			c.boolFlags[boolFlag.Name] = getValueForBoolFlag(boolFlag, baseContext)
+			if baseContext.IsSet(boolFlag.Name) {
```

Comment:
We probably want to always add value for bool flags, not only when they are set.